homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,364 @@
+    else:
+        persistent_notification.create(
+            hass, controller.error_message, "IntesisHome Error", 'intesishome')
+
+        controller.stop()
+        raise PlatformNotReady()
+
+
+class IntesisAC(ClimateDevice):
+    """Represents an Intesishome air conditioning device."""
+
+    def __init__(self, deviceid, ih_device, controller):
+        """Initialize the thermostat."""
+        _LOGGER.debug('Added climate device with state: %s', repr(ih_device))
+        self._controller = controller
+
+        self._deviceid = deviceid
+        self._devicename = ih_device.get('name')
+        self._has_swing_control = IH_SWING_WIDGET in ih_device.get('widgets')
+        self._connected = False
+        self._current_temp = None
+
+        self._max_temp = None
+        self._min_temp = None
+        self._target_temp = None
+
+        self._run_hours = None
+        self._rssi = None
+        self._swing = None
+        self._swing_list = None
+        self._vvane = None
+        self._hvane = None
+        self._power = False
+        self._fan_speed = STATE_UNKNOWN
+        self._hvac_mode = STATE_UNKNOWN
+        self._connection_retries = 0
+
+        self._hvac_modes = [HVAC_MODE_HEAT_COOL, HVAC_MODE_COOL,
+                            HVAC_MODE_HEAT, HVAC_MODE_DRY,
+                            HVAC_MODE_FAN_ONLY, HVAC_MODE_OFF]
+        self._fan_modes = [FAN_AUTO, IH_FAN_QUIET, FAN_LOW, FAN_MEDIUM,
+                           FAN_HIGH]
+
+        self._support = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE)
+
+        if self._has_swing_control:
+            self._support |= SUPPORT_SWING_MODE
+            self._swing_list = [SWING_OFF, SWING_BOTH,
+                                SWING_VERTICAL, SWING_HORIZONTAL]
+
+        self._controller.add_update_callback(self.update_callback)
+
+    @property
+    def name(self):
+        """Return the name of the AC device."""
+        return self._devicename
+
+    @property
+    def temperature_unit(self):
+        """Intesishome API uses celsius on the backend."""
+        return TEMP_CELSIUS
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        attrs = {}
+        if self._has_swing_control:
+            attrs['vertical_vane'] = self._vvane
+            attrs['horizontal_vane'] = self._hvane
+
+        if self._controller.is_connected:
+            attrs['ha_update_type'] = 'push'
+        else:
+            attrs['ha_update_type'] = 'poll'
+
+        return attrs
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        _LOGGER.debug("IntesisHome Set Temperature=%s")
+
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        hvac_mode = kwargs.get(ATTR_HVAC_MODE)
+
+        if hvac_mode:
+            self.set_hvac_mode(hvac_mode)
+
+        if temperature:
+            self._controller.set_temperature(self._deviceid, temperature)
```

Comment:
The asyncio transport callbacks are supposed to be run in the event loop. We should use the home assistant async api instead. So change to define `async def async_set_temperature(...):`.